static/js/pv_tree.tsx

```diff
@@ -0,0 +1,163 @@
+import React, {Component} from "react";
+import hierarchy from "../../tools/pv_tree_generator/hierarchy_golden.json";
+//import hierarchy from "../data/hierarchy.json";
+
+// Fields of each node from json file
+interface NodeType {
+  show: string; 
+  title: string; 
+  selected: string;
+  expanded: string;
+  children: NodeType[];
+  count: number;
+  search_count: number;
+  measuredProperty: string;
+  placeTypes: string[];
+  populationType: string;
+  type: string; // Property or Value
+  argString: string;// for statvars, argString = dcid
+}
+
+interface NodePropType {
+  title: string; 
+  selected: string;
+  expanded: string;
+  count: number;
+  children: NodeType[];
```

Comment:
Not quite sure how react handles the the memory of the Props, but the Menu object seems very heavy as each node contains the data for its subtree. Can you think if it is possible to create the tree without passing the data along the way?

Try https://reactjs.org/docs/react-api.html#createelement which takes the children elements as the third argument...

Reply:
Thinking a bit more I guess what you do is okay. Even all nodes are expanded, the total of all the props object is similar to the original object size. And if it's true that you only create the Node that is expanded, then the actual props passed down is very small. Creating node on the fly is definitely a good improvement!